lib/toPaths.js

```diff
@@ -1,13 +1,18 @@
 var isArray = Array.isArray;
 var typeOfObject = "object";
+var typeOfString = "string";
+var typeOfNumber = "number";
+var MAX_SAFE_INTEGER = 9007199254740991; // Number.MAX_SAFE_INTEGER in es6
+var abs = Math.abs;
+var safeNumberRegEx = /^(0|(\-?[1-9][0-9]*))$/;
```

Comment:
I'm just wondering if not treating 03 (for example) as a number is a conscious design decision or rather what the design motivation is for it. I'm not sure if we have it spec'd anywhere, so wondering if it's just a "that's probably what a user would expect" or a "not likely to happen" thing.